core/domain/question_jobs_one_off.py

```diff
@@ -118,3 +118,169 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, values)
+
+
+class QuestionSnapshotsMigrationAuditJob(jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-off job for testing the migration of all question
+    versions to the latest schema version. This job runs the state migration,
+    but does not commit the new question to the datastore.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [question_models.QuestionSnapshotContentModel]
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(QuestionSnapshotsMigrationAuditJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @staticmethod
+    def map(item):
+        question_id = item.get_unversioned_instance_id()
+
+        latest_question = question_services.get_question_by_id(
+            question_id, strict=False)
+        if latest_question is None:
+            yield ('INFO - Question does not exist', item.id)
+            return
+
+        question_model = question_models.QuestionModel.get(question_id)
+        if (question_model.question_state_data_schema_version !=
+                feconf.CURRENT_STATE_SCHEMA_VERSION):
+            yield (
+                'FAILURE - Question is not at latest schema version',
+                question_id)
+            return
+
+        try:
+            latest_question.validate()
+        except Exception as e:
+            yield (
+                'INFO - Question %s failed validation' % item.id,
+                e)
```

Comment:
Why would we update the schema of an invalid question? Shouldn't we leave it alone to improve debugability?

Reply:
We need to upgrade all the schemas if we are going to be able to successfully delete the earlier conversion functions from the codebase (which is the long-term aim of this PR).

Note that validate() only checks the latest version of the question. On a test run, I ran into exactly one case where validate() was failing because of repeated "invalid" content IDs in an ItemSelectionInput. This is a known issue and it does not affect the migratability of the snapshots. (Note that this was for the corresponding exploration-related code since we don't yet have any questions in prod.)

Reply:
OK, that sounds reasonable to me. Thanks!